src/server/core.js

```diff
@@ -11,23 +11,51 @@ function initTopics(topics) {
 
 function vote(state, topicId, voterId) {
   const voter = state.voters.find(voter => voter.id === voterId);
+
   if (!voter) {
-    state.topics[topicId].votes++;
-    state.voters.push({ id: voterId, vote: topicId });
+    const topics = Object.assign({}, state.topics, {
+      [topicId]: Object.assign({}, state.topics[topicId], {
```

Comment:
we could use already declared in upper scope variables topics/voters to not to declare them twice